### PR TITLE
Some improvement

### PR DESCRIPTION
**Pull request by samuelgarcia**
* Print sorters version.
  * log in json format inside folder
  * params in json format inside folder

### CODE REVIEW

**Comment by samuelgarcia** [comment]
@colehurwitz @alejoe91 : could you have a look to this.

Basically, I did:
  * print_sorter_version
  * refactor basesorter to force to dump params.json and log.json

after this I will need to make change in spiekcomparison too to retrieve run_times

**Comment by colehurwitz** [comment]
Looks good! A very useful tool and will address one of the reviewer's concerns (about reproducibility).

**Comment by alejoe91** [comment]
Hi @samuelgarcia 

Really nice! :) the params and the log in the base sorter are great!

Why is ironclust version hardcoded?

**Comment by alejoe91** [comment]
Also for `klusta` I think it would be better to report both `klusta` and `klustakwik2` versions

**Comment by samuelgarcia** [comment]
@alejoe91 for IC see this
ttps://github.com/flatironinstitute/ironclust/issues/39

I will have a look for klusta.

**Comment by colehurwitz** [comment]
Herdingspikes version 0.3.6 should have the version number now!

**Comment by alejoe91** [comment]
@samuelgarcia FYI if you run `irc` from the `ironclust/matlab` folder you get:
```
v5.0.5 (10/14/2019) installed, MD5: feba6fd83541b522f855c3e8a9cd1518
```

**Comment by samuelgarcia** [comment]
@alejoe91 : I would be happy to merge this soon.
I disable multiprocessing test. There is something strange with pytest.
I have some more change todo soon.

**Comment by samuelgarcia** [comment]
@alejoe91 @cole : can I merge this ?

File: spikesorters/basesorter.py [hunk 1]
```diff
@@ -79,10 +85,7 @@ def __init__(self, recording=None, output_folder=None, verbose=False,
                 raise RuntimeError(f"'{grouping_property}' is not one of the channel properties.")
             self.recording_list = recording.get_sub_extractors_by_property(grouping_property)
             n_group = len(self.recording_list)
-            if n_group > 1:
-                self.output_folders = [output_folder / str(i) for i in range(n_group)]
-            else:
-                self.output_folders = [output_folder / str(0)]
+            self.output_folders = [output_folder / str(i) for i in range(n_group)]
```
Comment (alejoe91) [inline spikesorters/basesorter.py]: nice :)

File: spikesorters/sorterlist.py [hunk 2]
```diff
@@ -93,6 +93,14 @@ def installed_sorters():
     '''
     return sorted(list([s.sorter_name for s in installed_sorter_list]))
 
+def print_sorter_version():
```
Comment (alejoe91) [inline spikesorters/sorterlist.py]: maybe `print_sorter_versions`? more than one sorter

Comment (samuelgarcia) [reply spikesorters/sorterlist.py]: If you want.

File: spikesorters/basesorter.py (hunk 1 above)
Comment (samuelgarcia) [reply spikesorters/basesorter.py]: This strange thing was from you.

**Pull request merged**